### PR TITLE
Add covariance method the FitResult.

### PR DESCRIPTION
**Pull request by marinang**
## Proposed Changes

Add covariance method in `FitResult`. Takes `params` as argument, and `as_dict` to return a dictionary.

## Tests added

None yet, Is there a place where the `FitResult` ( I didn't find it.) ? If not I will just create a test file.

### CODE REVIEW

File: tests/test_fitresult.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+@pytest.mark.parametrize("minimizer_class_and_kwargs", minimizers)
+def test_fmin(minimizer_class_and_kwargs):
+    results = create_fitresult(minimizer_class_and_kwargs=minimizer_class_and_kwargs)
+    result = results['result']
+    assert pytest.approx(results['cur_val']) == result.fmin
+
+
+@pytest.mark.parametrize("minimizer_class_and_kwargs", minimizers)
+def test_covariance(minimizer_class_and_kwargs):
+    pass
```
Comment (mayou36) [inline tests/test_fitresult.py]: I've added an explicit fitresult test file, feel free to put any test here. As an exapmle, see the `test_fmin` and the return from line 58

File: zfit/minimizers/fitresult.py [hunk 2]
```diff
@@ -239,6 +242,50 @@ def _error(self, params, method, sigma):
...
+                params (list(:py:class:`~zfit.Parameter`)): The parameters to calculate
+                    the covariance matrix. If `params` is `None`, use all *floating* parameters.
+                as_dict (bool): Default `False`. If `True` then returns a dictionnary.
+
+            Returns:
+                2D `numpy.array` of shape (N, N);
+                `dict`(param1, param2) -> covariance if `as_dict == True`.
+        """
+        params = self._input_check_params(params)
+        covariance_dict = self.minimizer._minuit_minimizer.covariance
```
Comment (mayou36) [inline zfit/minimizers/fitresult.py]: We need to be independent of minuit for sure. So for the moment, we can use the minuit covariance, but a standalone function should be there at some point (using _any_ library, we can use pure np for the calulations as well)

Can you put it into a try-except block catching the `AttributeError` and saying that only minuit covariance is currently there?

File: zfit/minimizers/fitresult.py [hunk 3]
```diff
@@ -239,6 +242,50 @@ def _error(self, params, method, sigma):
...
+def dict_to_matrix(params, matrix_dict):
+
+    nparams = len(params)
+    matrix = np.empty((nparams, nparams))
+
+    for i in range(nparams):
+        pi = params[i]
+        for j in range(i, nparams):
+            pj = params[j]
+            key = (pi, pj)
```
Comment (mayou36) [inline zfit/minimizers/fitresult.py]: I've refactored things slightly to always use the `Parameter` objects and not the string names.

Comment (marinang) [reply zfit/minimizers/fitresult.py]: Good, it makes sense to use the `Parameter`.

File: zfit/minimizers/fitresult.py (hunk 2 above)
Comment (marinang) [reply zfit/minimizers/fitresult.py]: Ok I agree but `MinuitMinimizer`doesn't have a covariance method. What's your idea of covariance implementation with other minimiser?

Comment (mayou36) [reply zfit/minimizers/fitresult.py]: Something similar to minuit for example. If I understand that correctly, it's an inversion of the error matrix, removal of the line and column that contains the parameter and a re-inversion.

Getting the error matrix, by hessian or by minos-like profiling, is another story, but also doable

Comment (marinang) [reply zfit/minimizers/fitresult.py]: Yeah it's something like this, like the inverse of the hessian matrix of the log-likelihood.

**Comment by marinang** [comment]
This should be good. The build failure comes from `test_complex_param`.

**Pull request merged**